cloud/services/domains.go

```diff
@@ -150,7 +153,18 @@ func (d *DNSEntries) getDNSEntriesToEnsure(mscope *scope.MachineScope) ([]DNSOpt
 	if mscope.LinodeMachine.Status.Addresses == nil {
 		return nil, fmt.Errorf("no addresses available on the LinodeMachine resource")
 	}
-	domainHostname := mscope.LinodeCluster.ObjectMeta.Name + "-" + mscope.LinodeCluster.Spec.Network.DNSUniqueIdentifier
+	clusterMetadata := mscope.LinodeCluster.ObjectMeta
+	clusterSpec := mscope.LinodeCluster.Spec
+	var domainHostname string
+	if clusterSpec.Network.DNSEndpointOverride != "" {
+		domainHostname = strings.ReplaceAll(clusterSpec.Network.DNSEndpointOverride, clusterSpec.Network.DNSRootDomain, "")
```

Comment:
is the expectation that the DNSEndpointOverride would contain the DNSRootDomain? It is fine if that is true, just want to make sure I understand what domainHostname is being created.

Reply:
post our discussion, i have updated the override field to only override the subdomain